pkg/network/ebpf/c/protocols/tls/go-tls-goid.h

```diff
@@ -39,10 +39,7 @@ static __always_inline int read_goroutine_id_from_tls(goroutine_id_metadata_t* m
         return 1;
     }
     void* goroutine_id_ptr = (void*) (g_addr + m->goroutine_id_offset);
-    if (bpf_probe_read(dest, sizeof(int64_t), goroutine_id_ptr)) {
-        return 1;
-    }
-    return 0;
+    return bpf_probe_read_user(dest, sizeof(int64_t), goroutine_id_ptr) < 0;
```

Comment:
Is there any specific reason not to use `return bpf_probe_read_user(dest, sizeof(int64_t), goroutine_id_ptr)`?

Reply:
yes, the function is expected to return 1 on failure and 0 on success, bpf_probe_read_user returns the opposite